third_party/move/move-prover/move-docgen/src/docgen.rs

```diff
@@ -1775,21 +1849,51 @@ impl<'env> Docgen<'env> {
 
     /// Outputs decorated code text in context of a module.
     fn code_text(&self, code: &str) {
-        emitln!(self.writer, &self.decorate_code(code));
+        if self.options.is_mdx_compatible() {
+            emit!(self.writer, "{}<br />", &self.decorate_code(code));
+        } else {
+            emitln!(self.writer, &self.decorate_code(code));
+        }
+    }
+
+    /// Replace html entities if the output format needs to be mdx compatible
+    fn replace_for_mdx(&self, cap: &Captures) -> String {
+        static MDX: Lazy<HashMap<&str, &str>> = Lazy::new(|| {
+            let mut map = HashMap::new();
+            map.insert("lt", "&lt;");
+            map.insert("gt", "&gt;");
+            map.insert("lb", "&#123;");
+            map.insert("rb", "&#125;");
+            map.insert("amper", "&amp;");
+            map.insert("dquote", "&quot;");
+            map.insert("squote", "&apos;");
+            map.insert("sharp", "&#35;");
+            map.insert("mul", "&#42;");
+            map.insert("plus", "&#43;");
+            map.insert("minus", "&#45;");
+            map.insert("eq", "&#61;");
+            map.insert("bar", "&#124;");
+            map.insert("tilde", "&#126;");
+            map.insert("nl", "<br />");
+            map
+        });
+        let mut r = "".to_string();
+        for (group_name, replacement) in MDX.iter() {
```

Comment:
Wow, this seems really inefficient, but I can't see a better way with the Regexp methods.

You probably want something other than an `HashMap` above, since you're using the map as a list of pairs.  How about a vector?